### PR TITLE
Add initial support for new board - Adafruit KB2040

### PR DESCRIPTION
**Pull request by markbirss**
Initial support for new board Adafruit KB2040

Kindly review and comment

### CODE REVIEW

File: source/pintab_adafruit_kb2040.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from board import (
+    A3,
```
Comment (bill88t) [inline source/pintab_adafruit_kb2040.py:2]: Huh, `A`-pins? Well, thankfully the pintab exists just for that.

File: rootfilesystem/config-adafruit_kb2040.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "_comment1": "Default Ljinux configuration, adjust to your needs",
+  
+  "_comment2": "As long as it's true, ram rtc will be used",
+  "fixrtc": true,
+  
+  "_comment3": "Board led, can be generic or neopixel",
+  "led": 17,
+  "ledtype": "generic",
```
Comment (bill88t) [inline rootfilesystem/config-adafruit_kb2040.json:9]: Generic? Is it not the neopixel?

**Review by bill88t — COMMENTED** [review]
Looks good, only the led seems wrong.
`generic` is the type of led found on the Raspberry Pi Pico, it only needs power, no logic.
`neopixel` is for neopixels.

File: rootfilesystem/config-adafruit_kb2040.json (hunk 2 above)
Comment (markbirss) [reply rootfilesystem/config-adafruit_kb2040.json:9]: Yes, it is a neopixel

File: source/pintab_adafruit_kb2040.py (hunk 1 above)
Comment (markbirss) [reply source/pintab_adafruit_kb2040.py:2]: Yes, I took all of them from the diagram provided as pinout

**Comment by bill88t** [comment]
Changed the pin specified in the json, to match the neopixel number (9) in accordance to the pintab.
Should you try to boot ljinux with these changes, you should see quite a lot of activity on that neopixel.
Please verify it works.

**Comment by markbirss** [comment]
@bill88t im not sure if the pintab numbering is all correct ? The neopixel is on GPIO17

File: source/pintab_adafruit_kb2040.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    A2,
+    A1,
+    A0,
+    SCK,     
+    MISO,
+    MOSI,
+    D10,   
+    NEOPIXEL,
```
Comment (bill88t) [inline source/pintab_adafruit_kb2040.py:10]: This imports `board.NEOPIXEL`.

File: source/pintab_adafruit_kb2040.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+pintab = {
+    1: A3,
+    2: A2,
+    3: A1,
+    4: A0,
+    5: SCK,
+    6: MISO,
+    7: MOSI,
+    8: D10,
+    9: NEOPIXEL,
```
Comment (bill88t) [inline source/pintab_adafruit_kb2040.py:35]: And this maps the `NEOPIXEL` as GP9. And the json relies on this decleration.
Or is this not the one we are to use as the act led?

**Review by bill88t — COMMENTED** [review]
Pinpoined the declaration in question

**Comment by bill88t** [comment]
> @bill88t im not sure if the pintab numbering is all correct ? The neopixel is on GPIO17 https://learn.adafruit.com/adafruit-kb2040/pinouts

Is it `GPIO17` or `NEOPIXEL` in the board module?
The `"led"` value is determined in accordance to the pintab, not based on real GPIO, and the pintab you gave set it to value 9.

**Comment by bill88t** [comment]
Adafruit says it's board.NEOPIXEL (https://learn.adafruit.com/adafruit-kb2040/pinouts#neopixel-led-3106797).
Which is what you set as 9 in the pintab, so I am pretty sure the led is correct as is.

**Review by bill88t — COMMENTED** [review]
Yea, this ordering scheme makes more sense, but shouldn't `A0`-`A3` allocate 10-13?

**Comment by markbirss** [comment]
> Yea, this ordering scheme makes more sense, but shouldn't `A0`-`A3` allocate 10-13?

I have taken the pinmap from this diagram, please advice
image

**Comment by markbirss** [comment]
Ok, thank you

**Comment by markbirss** [comment]
Let me do a quick test again

**Comment by bill88t** [comment]
> I have taken the pinmap from this diagram, please advise

Oh alright then. I agree it's best to stick to the pinout since we can.
Now imma make the pinout command's ascii art.

**Comment by markbirss** [comment]
Ok, i tested, and neopixel much more activity now

**Comment by bill88t** [comment]
Alrighty then, as soon as I get the ascii pinout done, I'm gonna merge it in.
Cuz atm, the pinout command prolly will crash.

**Comment by markbirss** [comment]
@bill88t Ok, thank you, great project

**Pull request merged**